src/ipumspy/api/extract.py

```diff
@@ -172,6 +177,42 @@ def extract_from_dict(dct: Dict[str, Any]) -> Union[BaseExtract, List[BaseExtrac
         return [extract_from_dict(extract) for extract in dct["extracts"]]
 
     if dct["collection"] in BaseExtract._collection_to_extract:
+        # cosmetic procedure for when dct comes from json file
+        for key in ["samples", "variables"]:
+            if type(dct[key]) is dict:
```

Comment:
It's generally better to use `isinstance`, so:

```python
if isinstance(dct[key], dict):
    dct[key] = list(dct[key].keys())
```

Reply:
ah - yes, thank you for the reminder. Old (bad) habits die hard.